### PR TITLE
Added ignored RoPE block

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- As stated in the title.
- Added `Sin` and `Cos` OV operations.

### Reason for changes

- Accuracy improvement for models with `ModelType.TRANSFORMERS` type.

### Related tickets

- 155511

### Tests

- Added graph test for OV.

image

### CODE REVIEW

**Comment by KodiaqQ** [comment]
@openvinotoolkit/nncf-maintainers, can someone review this, please. Thank you.
All detailed other was placed in the ticket.

File: tests/torch/test_pattern_manager.py [hunk 1]
```diff
@@ -74,6 +74,7 @@
 IGNORING_IGNORED_PATTERN_REASONS = {
     IgnoredPatternNames.FC_BN_HSWISH_ACTIVATION: "Not relevant for Torch.",
     IgnoredPatternNames.EQUAL_LOGICALNOT: "Not relevant for Torch.",
+    IgnoredPatternNames.ROPE: "Not relevant for Torch.",
```
Comment (alexsu52) [inline tests/torch/test_pattern_manager.py:77]: Could you explain why this pattern is not relevant for other backends?

Comment (KodiaqQ) [reply tests/torch/test_pattern_manager.py:77]: There are no examples of this pattern in other backends. I can not add this for ONNX/PT since I have no models with this structure from these frameworks.

Comment (alexsu52) [reply tests/torch/test_pattern_manager.py:77]: What models did you use for testing the RoPE pattern for OV backend? If source models are PyTorch models you can reuse them to specify RoPE pattern for ONNX/PT backends.

Comment (KodiaqQ) [reply tests/torch/test_pattern_manager.py:77]: Ok

File: tests/cross_fw/test_templates/helpers.py [hunk 2]
```diff
@@ -466,3 +466,22 @@ def __init__(self):
 
     def forward(self, x):
         return self.conv(x)
+
+
+class RoPEModel(nn.Module):
```
Comment (KodiaqQ) [inline tests/cross_fw/test_templates/helpers.py:471]: I wasn't able to clearly generate the PT graph to properly visualise the pattern variant. I've created a model based on this interpretation (https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L149-L169) since I have no others.
@alexsu52

File: tests/torch/test_pattern_manager.py (hunk 1 above)
Comment (KodiaqQ) [reply tests/torch/test_pattern_manager.py]: Added patterns with the corresponding tests as you asked. Need to properly review it, since I've had to create it without strong ONNX/PT research due to lack of time.

File: nncf/onnx/quantization/ignored_patterns.py [hunk 3]
```diff
@@ -157,3 +157,25 @@ def create_se_block() -> GraphPattern:
                 multiply_node = node_id
         pattern.add_edge(non_pattern_node, multiply_node)
     return pattern
+
+
+@ONNX_IGNORED_PATTERNS.register(IgnoredPatternNames.ROPE)
+def create_rope() -> GraphPattern:
```
Comment (alexsu52) [inline nncf/onnx/quantization/ignored_patterns.py:163]: @kshpv, please review this part

File: nncf/torch/quantization/ignored_patterns.py [hunk 4]
```diff
@@ -227,3 +227,23 @@ def get_se_block_with_bias_and_reshape() -> GraphPattern:
     main_pattern.add_pattern_alternative(get_se_block_with_reshape())
     main_pattern.add_pattern_alternative(get_se_block_with_bias_and_reshape())
     return main_pattern
+
+
+@PT_IGNORED_PATTERNS.register(IgnoredPatternNames.ROPE)
+def create_rope() -> GraphPattern:
```
Comment (alexsu52) [inline nncf/torch/quantization/ignored_patterns.py:233]: @AlexanderDokuchaev, please review this pattern.

File: tests/cross_fw/test_templates/helpers.py (hunk 2 above)
Comment (alexsu52) [reply tests/cross_fw/test_templates/helpers.py:471]: Thanks, looks good.

File: tests/torch/test_pattern_manager.py (hunk 1 above)
Comment (alexsu52) [reply tests/torch/test_pattern_manager.py]: On which models have you tested the pattern?

**Comment by MaximProshin** [comment]
@KodiaqQ , @alexsu52 , let's split it into 2 PRs: first for OV and second for other backends.

**Comment by KodiaqQ** [comment]
> @KodiaqQ , @alexsu52 , let's split it into 2 PRs: first for OV and second for other backends.

https://github.com/openvinotoolkit/nncf/pull/3059

**Review by alexsu52 — APPROVED** [review]
LGTM
